accord-core/src/main/java/accord/primitives/Routables.java

```diff
@@ -55,62 +81,147 @@
     /**
      * Perform {@link SortedArrays#exponentialSearch} from {@code thisIndex} looking for {@code find} with behaviour of {@code search}
      */
-    int findNext(int thisIndex, K find, SortedArrays.Search search);
+    int findNext(int thisIndex, RoutableKey find, SortedArrays.Search search);
+
+    Domain domain();
+
+    @Inline
+    static <Param, Output> Output switchOnDomain(Routables<?, ?> inputs, Param param, BiFunction<AbstractKeys<?, ?>, Param, Output> ifKeys, BiFunction<AbstractRanges<?>, Param, Output> ifRanges)
+    {
+        switch (inputs.domain())
+        {
+            default: throw new AssertionError();
+            case Key: return ifKeys.apply((AbstractKeys<?, ?>) inputs, param);
+            case Range: return ifRanges.apply((AbstractRanges<?>) inputs, param);
+        }
+    }
 
-    Routable.Domain kindOfContents();
+    @Inline
+    static <Param, Output> Output switchOnDomain(Domain domain, Param param, Function<Param, Output> ifKeys, Function<Param, Output> ifRanges)
```

Comment:
This is used once to produce either an empty `PartialKeyRoute` or `PartialRangeRoute`. Seems like it could just be a default method `toPartialRoute` in the `PartialRouteInterface`?

Reply:
Do you mean a static method?

Reply:
Oops yes